docs/internal/GeneralArchitectureGuide.md

```diff
@@ -101,6 +101,24 @@ are coordinated.
 > does not hold, in those cases you can locate the transport action for a REST action by looking at the `NodeClient` invocation in the
 > `Rest*Action`'s `prepareRequest` implementation, it should specify the `ActionType` being invoked which can then be used to locate
 > the `Transport*Action` class that handles it.
+>
+> A netty [EventLoop] thread handles the initial steps of a Rest*Action request lifecycle such as decoding, validation and routing.
+> Upon entry into the "transport layer", [NodeClient] delegates the decision of execution to individual [TransportAction]. Each action
```

Comment:
Personally, I prefer not to make every mention of a class name a code link. Maybe use a link the first time a class is mentioned, if you think it's helpful to go look at some particular piece of code for additional information.

When I see a link, I think I should click on it, and then I'm disappointed when it's redundant and I have too many tabs open (the last part is my own problem I suppose 😌).

Instead of highlighting text with a hyperlink, you can use back ticks (``) to emphasize that it's a code name. I'd replace all your new text's [TransportAction] with `TransportAction`, since there's already a link reference above the new text. Same for [ActionType]